### PR TITLE
Fixup privatebin

### PR DESCRIPTION
**Pull request by khazhyk**

### CODE REVIEW

**Comment by khazhyk** [comment]
@iomintz

**Comment by iomintz** [comment]
What does this ping mean? Would you like me to test it, review it?

**Comment by khazhyk** [comment]
review, idk. how did you test ur impl before

**Comment by iomintz** [comment]
Hmm… don't remember. I think I just ran it…

```py
🐍 asyncio.run(privatebin.upload('foo'))
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/lib/python3.7/asyncio/runners.py", line 43, in run
    return loop.run_until_complete(main)
  File "/usr/lib/python3.7/asyncio/base_events.py", line 579, in run_until_complete
    return future.result()
  File "/home/io/code/vcs/git/com/github/@/khazhyk/dango.py/dango/privatebin.py", line 84, in upload
    raise PrivateBinException(resp, await resp.text())
dango.privatebin.PrivateBinException: (<ClientResponse(https://privatebin.net/) [500 Internal Server Error]>
<CIMultiDictProxy('Server': 'nginx/1.17.3', 'Date': 'Wed, 16 Oct 2019 07:43:17 GMT', 'Content-Type': 'text/html; charset=UTF-8', 'Transfer-Encoding': 'chunked', 'Connection': 'keep-alive')>
```
anyway this is what i get now

**Comment by iomintz** [comment]
```
curl \
    'https://privatebin.net/' \
    -H 'User-Agent: Mozilla/5.0 (X11; Linux x86_64; rv:69.0) Gecko/20100101 Firefox/69.0' \
    -H 'Accept: application/json, text/javascript, */*; q=0.01' \
    -H 'Accept-Language: en-US,en;q=0.5' \
    --compressed \
    -H 'Content-Type: application/x-www-form-urlencoded; charset=UTF-8' \
    -H 'X-Requested-With: JSONHttpRequest' \
    -H 'DNT: 1' \
    -H 'Connection: keep-alive' \
    -H 'Pragma: no-cache' \
    -H 'Cache-Control: no-cache' \
    -H 'TE: Trailers' \
    --data '{"adata":[["8kSZyKmWADZPHbpS4P6SJQ==","5GzPsMqCa3A=",100000,256,128,"aes","gcm","zlib"],"plaintext",0,0],"meta":{"expire":"1week"},"v":2,"ct":"6xpM6k+pRu3Cn3dsSPZgaaztOagHgcJlvwCYtli1RNPaDA=="}'
```
this is what my browser generated for a test upload. it honestly _looks_ the same.

and i remember how i tested my impl. against the one i copied from, https://github.com/r4sas/PBinCLI. I'll see how it does things nowadays.

File: dango/privatebin.py [hunk 1]
```diff
@@ -34,22 +34,31 @@ def compress(src: bytes):
...
         cipher[k] = cipher[k].decode()
 
-    request['data'] = json.dumps(cipher, ensure_ascii=False,
-                                 indent=None, default=lambda x: x.decode('utf-8'))
+    request["adata"] = [
+        [
+            cipher["iv"], cipher["salt"], cipher["iter"], cipher["ks"],
+            cipher["ts"], cipher["cipher"], cipher["mode"], "zlib"],
+        "plaintext", 0, 0
+    ]
```
Comment (iomintz) [inline dango/privatebin.py]: replace this with an assignment to `request["adata"][0]`

**Pull request closed without merging**